From 8827db53bf2d12c7c0233f27f3bd082564894e6b Mon Sep 17 00:00:00 2001
From: Chun-Yi Lee <jlee@suse.com>
Date: Thu, 19 Dec 2024 15:56:20 +0800
Subject: [PATCH 3/5] efi: Set early kernel lock down flag if booted in secure
 boot mode
Patch-mainline: Never, temporary solution
References: bsc#1234646

After '77b644c39d6a init/main.c: Initialize early LSMs after arch code,
static keys and calls' be introduced in v6.12, the lockdown LSM does
not work in setup_arch() now. This patch set a early kernel lock down
flag for filling the gap.

Signed-off-by: Chun-Yi Lee <jlee@suse.com>
---
 arch/x86/kernel/setup.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kernel/setup.c b/arch/x86/kernel/setup.c
index f6e985f64..2519960a1 100644
--- a/arch/x86/kernel/setup.c
+++ b/arch/x86/kernel/setup.c
@@ -908,8 +908,10 @@ void __init setup_arch(char **cmdline_p)
 	efi_set_secure_boot(boot_params.secure_boot);
 
 #ifdef CONFIG_LOCK_DOWN_IN_EFI_SECURE_BOOT
-	if (efi_enabled(EFI_SECURE_BOOT))
+	if (efi_enabled(EFI_SECURE_BOOT)) {
 		security_lock_kernel_down("EFI Secure Boot mode", LOCKDOWN_INTEGRITY_MAX);
+		lock_kernel_down_early("EFI Secure Boot mode", LOCKDOWN_INTEGRITY_MAX);
+	}
 #endif
 
 	reserve_ibft_region();
-- 
2.35.3

